stattest/experiment/generator/generator_step.py

```diff
@@ -24,55 +20,49 @@ def generate_rvs_data(rvs_generator: AbstractRVSGenerator, size, count):
     :return: Data Frame, where rows is rvs
     """
 
-    return [rvs_generator.generate(size) for i in range(count)]
+    return [rvs_generator.generate(size) for _ in range(count)]
 
 
-def prepare_one_size_rvs_data(
-    rvs_generators: [AbstractRVSGenerator], size, count, store: IRvsStore, pbar=None
+def process_entries(
+    generate_queue: Queue,
+    info_queue: Queue,
+    generate_shutdown_event: Event,
+    info_shutdown_event: Event,
+    kwargs,
 ):
-    """
-    Generate data rvs and save it to store.
-
-    :param size: size of rvs
-    :param store: store to persist data
-    :param rvs_generators: generators list to generate rvs data
-    :param count: rvs count
-    """
+    store = kwargs["store"]
+    store.init()
 
-    for generator in rvs_generators:
-        code = generator.code()
-        data_count = store.get_rvs_count(code, size)
-        if data_count < count:
-            count = count - data_count
+    while not (generate_shutdown_event.is_set() and generate_queue.empty()):
+        if not generate_queue.empty():
+            generator, size, count = generate_queue.get()
             data = generate_rvs_data(generator, size, count)
-            store.insert_all_rvs(code, size, data)
-        if pbar:
-            pbar.update(1)
+            store.insert_all_rvs(generator.code(), size, data)
+            info_queue.put(1)
 
+    info_shutdown_event.set()
 
-def prepare_rvs_data(
-    rvs_generators: [AbstractRVSGenerator],
-    sizes,
-    count,
-    store: IRvsStore,
-    thread_count: int = 0,
-):
-    """
-    Generate data rvs and save it to store.
 
-    :param rvs_generators: generators list to generate rvs data
-    :param sizes: sizes of rvs
-    :param store: store to persist data
-    :param count: rvs count
-    """
+def fill_queue(
+    queue,
+    generate_shutdown_event,
+    sizes=None,
+    count=0,
+    store=None,
+    rvs_generators: [AbstractRVSGenerator] = None,
+    **kwargs,
```

Comment:
```sizes```, ```count```, ```store```, ```rvs_generators``` parameters are not passed from ```start_pipeline``` function neither explicitly, nor from ```**kwargs```. Does it work a little more tricky?